src/gentropy/datasource/gwas_catalog/study_index.py

```diff
@@ -264,14 +265,14 @@ def parse_cohorts(raw_cohort: Column) -> Column:
         |parsedCohorts                         |
         +--------------------------------------+
         |[BioME, CaPS, Estonia, FHS, UKB, GERA]|
-        |[null]                                |
+        |null                                  |
         +--------------------------------------+
         <BLANKLINE>
         """
         return f.when(
-            (raw_cohort.isNull()) | (raw_cohort == ""),
-            f.array(f.lit(None).cast(t.StringType())),
-        ).otherwise(f.array_distinct(f.split(raw_cohort, r"\|")))
+            (raw_cohort.isNotNull()) & (raw_cohort != ""),
```

Comment:
This update was required to make sure the resulting dataset would not contain arrays with nulls.